spec/requests/users_spec.rb

```diff
@@ -0,0 +1,107 @@
+require 'rails_helper'
+
+RSpec.describe 'Users API', type: :request do
+  let!(:users) { create_list(:user, 10) }
+  let(:user_id) { users.first.id }
+
+  # Test suite for GET /api/users
+  describe 'GET /api/users' do
+    # make HTTP get request before each example
+    before { get '/api/users' }
+
+    it 'returns users' do
+      # Note `json` is a custom helper to parse JSON responses
+      expect(json).not_to be_empty
+      expect(json.size).to eq(10)
+    end
+
+    it 'returns status code 200' do
+      expect(response).to have_http_status(200)
+    end
+  end
+
+  # Test suite for GET /api/users/:id
+  describe 'GET /api/users/:id' do
+    before { get "/api/users/#{user_id}" }
+
+    context 'when the record exists' do
+      it 'returns the todo' do
+        expect(json).not_to be_empty
+        expect(json['id']).to eq(user_id)
+      end
+
+      it 'returns status code 200' do
+        expect(response).to have_http_status(200)
+      end
+    end
+
+    context 'when the record does not exist' do
+      let(:user_id) { 100 }
+
+      it 'returns status code 404' do
+        expect(response).to have_http_status(404)
+      end
+
+      it 'returns a not found message' do
+        expect(response.body).to match(/Couldn't find User/)
+      end
+    end
+  end
+
+  # Test suite for POST /api/users
+  describe 'POST /api/users' do
+    # valid payload
+    let(:valid_attributes) { { first_name: 'Rogelio', last_name: 'Alatorre', email: 'roger@apimon.com', mobile_no: '3122111436' } }
+
+    context 'when the request is valid' do
+      before { post '/api/users', params: valid_attributes }
+
+      it 'creates a user' do
+        expect(json['first_name']).to eq('Rogelio')
```

Comment:
Add scenarios to retrieve the error messages when some of the attributes are missing on the parameters.